src/bridge/test-metrics.c

```diff
@@ -850,6 +850,58 @@ test_cpu_cores (void)
   g_object_unref (transport);
 }
 
+static void
+test_cpu_temperature (void)
+{
+  MockTransport *transport = mock_transport_new ();
+
+  g_autoptr (JsonObject) options = json_obj ("{ 'metrics': [ { 'name': 'cpu.temperature' } ],"
+                                  "  'interval': 1000"
+                                  "}");
+
+  g_signal_connect (transport, "closed", G_CALLBACK (on_transport_closed), NULL);
+
+  CockpitChannel *channel = g_object_new (cockpit_internal_metrics_get_type (),
+                          "transport", transport,
+                          "id", "1234",
+                          "options", options,
+                          NULL);
+
+  cockpit_channel_prepare (channel);
+
+  g_autoptr (GBytes) msg;
+  /* receive meta information */
+  while ((msg = mock_transport_pop_channel (transport, "1234")) == NULL)
+    g_main_context_iteration (NULL, TRUE);
+
+  g_autoptr (GError) error = NULL;
+  g_autoptr (JsonObject) res = cockpit_json_parse_bytes (msg, &error);
+  g_assert_no_error (error);
+  g_assert (res != NULL);
+  g_assert (json_object_has_member (res, "metrics"));
+
+  JsonArray *metrics = json_object_get_array_member (res, "metrics");
```

Comment:
To avoid double (or more) frees on same memory. Here `metrics` points to a part of memory which is freed with `g_autoptr(JsonObject) res` at line 878.

Reply:
These should ideally be `const` then. Sometimes this isn't possible as APIs are not appropriately `const`ified, but whenever you can, please do.

Reply:
json-glib is not constified at all, so nevermind.